### PR TITLE
Help Command: add the ability to request help for a specific command

### LINKED ISSUE #3: As a new user, I can view a help page / use a help command
...so that I know the correct syntax to use the functions / know what functions the app supports

### PR DESCRIPTION
**Pull request by nickyfoo**
Add the ability for users to request help for a single command
Show all the syntax guides directly in the application when requested instead of having to open a new link

Update UserGuide on Help command functionality and syntax

Fixes #3

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 1]
```diff
@@ -12,10 +12,73 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows program usage instructions.\n"
             + "Example: " + COMMAND_WORD;
 
-    public static final String SHOWING_HELP_MESSAGE = "Opened help window.";
+    public static final String SHOWING_HELP_MESSAGE = "Here is a list of commands available, "
+            + " for more information refer to https://ay2021s2-cs2103-w16-3.github.io/tp/UserGuide.html\n"
```
Comment (pyuxiang) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java:16]: Extra space.

```suggestion
            + "for more information refer to https://ay2021s2-cs2103-w16-3.github.io/tp/UserGuide.html\n"
```

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 2]
```diff
@@ -12,10 +12,73 @@
...
+            + "Editing contacts: "
+            + "edit INDEX [-n NAME] [-p PHONE_NUMBER] [-e EMAIL] [-a ADDRESS] [-t TAG]…\u200B [-b BIRTHDAY]\n"
+            + "Finding contacts: "
+            + "find [-n NAME] [-t TAG]\n"
+            + "Listing contacts: "
+            + "list [-s SORT_ORDER]\n"
+            + "Finding tags: "
+            + "tags [-f KEYWORD]\n"
+            + "Showing help: "
+            + "help [COMMAND]\n";
```
Comment (pyuxiang) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java:31]: Extra newline.

```suggestion
            + "help [COMMAND]";
```

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 3]
```diff
@@ -12,10 +12,73 @@
...
-    public static final String SHOWING_HELP_MESSAGE = "Opened help window.";
+    public static final String SHOWING_HELP_MESSAGE = "Here is a list of commands available, "
+            + " for more information refer to https://ay2021s2-cs2103-w16-3.github.io/tp/UserGuide.html\n"
+            + "Adding contacts: "
+            + "add NAME [-p PHONE_NUM] [-e EMAIL] [-a ADDRESS] [-t TAG]…\u200B [-b BIRTHDAY]\u200B\n"
+            + "Clearing all data: clear\n"
+            + "Deleting contacts: "
+            + "delete INDEX [INDEX…]\n"
+            + "Editing contacts: "
+            + "edit INDEX [-n NAME] [-p PHONE_NUMBER] [-e EMAIL] [-a ADDRESS] [-t TAG]…\u200B [-b BIRTHDAY]\n"
```
Comment (pyuxiang) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java:23]: Missing `exit` command.

```suggestion
            + "edit INDEX [-n NAME] [-p PHONE_NUMBER] [-e EMAIL] [-a ADDRESS] [-t TAG]…\u200B [-b BIRTHDAY]\n"
            + "Exiting application: "
            + "exit\n"
```

**Review by pyuxiang — CHANGES_REQUESTED** [review]
Some other comments:

- ``help exit`` and ``help clear`` currently do not have examples for usage, good idea to add in? i.e. `Example: clear`

File: docs/UserGuide.md [hunk 4]
```diff
@@ -81,7 +81,7 @@ Examples:
 
 Edits an existing person in the HippoCamus' Contact List.
 
-Format: `edit INDEX [-n NAME] [-p PHONE_NUMBER] [-e EMAIL] [-a ADDRESS] [-t TAG…​ [-b BIRTHDAY]`
+Format: `edit INDEX [-n NAME] [-p PHONE_NUMBER] [-e EMAIL] [-a ADDRESS] [-t TAG]…​ [-b BIRTHDAY]`
```
Comment (garyljj) [inline docs/UserGuide.md:84]: should be [-t TAG...]

File: src/main/java/seedu/address/logic/commands/HelpCommand.java (hunk 3 above)
Comment (garyljj) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java:23]: same here, should be [-t TAG...]
and im not sure whats that special char. but if its intentional then no problem!

**Review by garyljj — CHANGES_REQUESTED** [review]
Just a very minor typo
Edit: resolved, not a typo

File: docs/UserGuide.md (hunk 4 above)
Comment (garyljj) [reply docs/UserGuide.md:84]: Cancel, not a typo

File: src/main/java/seedu/address/logic/commands/HelpCommand.java (hunk 1 above)
Comment (nickyfoo) [reply src/main/java/seedu/address/logic/commands/HelpCommand.java:16]: Thanks, corrected.

File: src/main/java/seedu/address/logic/commands/HelpCommand.java (hunk 3 above)
Comment (nickyfoo) [reply src/main/java/seedu/address/logic/commands/HelpCommand.java:23]: Added, thanks

**Review by garyljj — APPROVED** [review]
LGTM

**Review by pyuxiang — APPROVED** [review]
LGTM

**Pull request merged**